File: Regression/Section 8 - Decision Tree Regression/decisionTreeRegressionSolution.R
# Decision Tree Regression

# Importing the dataset
dataSet = read.csv('Position_Salaries.csv')
dataSet = dataSet[2:3]

# Fitting Decision Tree Regression to the dataset
#install.packages('rpart')
library(rpart)
regressor = rpart(formula = Salary ~ .,
                  data = dataSet,
                  control = rpart.control(minsplit = 1))

# Predicting a new result with Decision Tree Regression
y_pred = predict(regressor, data.frame(Level = 6.5))

# Visualising the Decision Tree Regression results (higher resolution)
# install.packages('ggplot2')
library(ggplot2)
xGrid = seq(min(dataSet$Level), max(dataSet$Level), 0.01)
ggplot() +
  geom_point(aes(x = dataSet$Level, y = dataSet$Salary),
             colour = 'red') +
  geom_line(aes(x = xGrid, y = predict(regressor, newdata = data.frame(Level = xGrid))),
            colour = 'blue') +
  ggtitle('Truth or Bluff (Decision Tree Regression)') +
  xlab('Level') +
  ylab('Salary')

# Plotting the tree
plot(regressor)
text(regressor)

File: Regression/Section 8 - Decision Tree Regression/decisionTreeRegressionSolution.py
# dECISION tREE rEGRESSION

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataSet = pd.read_csv('Position_Salaries.csv')
X = dataSet.iloc[:, 1:2].values
y = dataSet.iloc[:, 2].values

# Fitting the Regression Model to the dataset
from sklearn.tree import DecisionTreeRegressor
regressor = DecisionTreeRegressor(random_state = 0)
regressor.fit(X, y)

# Predicting a new result
yPredict = regressor.predict(6.5)

# Visualising the Regression results
XGrid = np.arange(min(X), max(X), 0.01)
XGrid = XGrid.reshape((len(XGrid), 1))
plt.scatter(X, y, color = 'red')
plt.plot(XGrid, regressor.predict(XGrid), color = 'blue')
plt.title('Truth or Bluff (Regression Model)')
plt.xlabel('Position level')
plt.ylabel('Salary')
plt.show()
